In [129]:
import pandas as pd

In [131]:
df = pd.read_excel("Financial_Summary_Microsoft_Tesla_Apple.xlsx")

In [133]:
df.head()

Company  Fiscal Year  Total Revenue  Net Income   Total Assets   \
0  Microsoft         2024         245.10        88.10          512.0   
1  Microsoft         2023         211.90        72.40          411.0   
2  Microsoft         2022         198.30        72.70          364.8   
3      Tesla         2024          97.69         7.09          102.0   
4      Tesla         2023          96.77        14.99           94.0   

   Total Liabilities   Operating Cash Flow  
0               198.3                95.20  
1               183.0                89.00  
2               176.1                83.40  
3                38.0                14.92  
4                35.0                13.26

#### Note :
The data is in billion dollar format

In [135]:
df.columns

Index(['Company', 'Fiscal Year', 'Total Revenue', 'Net Income ',
       'Total Assets ', 'Total Liabilities ', 'Operating Cash Flow'],
      dtype='object')

In [137]:
df = df.sort_values(by = ['Company','Fiscal Year']).reset_index()

In [148]:
df.drop('index',axis = 1,inplace = True)

In [150]:
df['Revenue Growth (%)'] =  df.groupby(['Company'])['Total Revenue'].pct_change()*100

In [152]:
df['Net Income  Growth (%)'] =  df.groupby(['Company'])['Net Income '].pct_change()*100

In [154]:
df

Company  Fiscal Year  Total Revenue  Net Income   Total Assets   \
0      Apple         2022         394.30        99.80          352.8   
1      Apple         2023         383.30        96.90          352.8   
2      Apple         2024         391.00        93.70          352.8   
3  Microsoft         2022         198.30        72.70          364.8   
4  Microsoft         2023         211.90        72.40          411.0   
5  Microsoft         2024         245.10        88.10          512.0   
6      Tesla         2022          81.46        12.56           82.0   
7      Tesla         2023          96.77        14.99           94.0   
8      Tesla         2024          97.69         7.09          102.0   

   Total Liabilities   Operating Cash Flow  Revenue Growth (%)  \
0               290.6               122.20                 NaN   
1               290.6               110.50           -2.789754   
2               287.0               110.50            2.008870   
3               176.1                83.40                 NaN   
4               183.0                89.00            6.858296   
5               198.3                95.20           15.667768   
6                33.0                14.72                 NaN   
7                35.0                13.26           18.794500   
8                38.0                14.92            0.950708   

   Net Income  Growth (%)  
0                     NaN  
1               -2.905812  
2               -3.302374  
3                     NaN  
4               -0.412655  
5               21.685083  
6                     NaN  
7               19.347134  
8              -52.701801

# Revenue Growth and Net Income Growth Analysis

## Apple
- In **2023**, Apple's revenue declined by **2.79%**, and net income decreased by **2.91%**.
- In **2024**, revenue slightly recovered with a **2.01%** growth, but net income continued declining by **3.30%**.
- The continuous drop in net income suggests increasing costs or shrinking profit margins, despite revenue stabilization.

## Microsoft
- In **2023**, Microsoft saw a **6.86%** increase in revenue, but net income slightly dropped by **0.41%**.
- In **2024**, revenue surged by **15.67%**, accompanied by a strong net income growth of **21.69%**.
- This indicates Microsoft's ability to drive higher profitability alongside revenue expansion, potentially due to efficiency improvements or successful business strategies.

## Tesla
- In **2023**, Tesla experienced a **18.79%** revenue increase, alongside a **19.35%** rise in net income.
- However, in **2024**, revenue growth slowed dramatically to **0.95%**, while net income plunged by **52.70%**.
- The sharp decline in net income suggests major profitability challenges, possibly linked to operational costs or external market factors.

